nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/RHostedFacetImpl.java

```diff
@@ -66,57 +70,13 @@
     extends FacetSupport
     implements RHostedFacet
 {
-  @Override
-  @TransactionalTouchMetadata
-  public Content getPackages(final String packagesPath) {
-    checkNotNull(packagesPath);
-    try {
-      // TODO: Do NOT do this on each request as there is at least some overhead, and memory usage is proportional to
-      // the number of packages contained in a particular path. We should be able to generate this when something has
-      // changed or via a scheduled task using an approach similar to the Yum implementation rather than this method.
-      StorageTx tx = UnitOfWork.currentTx();
-      RPackagesBuilder packagesBuilder = new RPackagesBuilder(packagesPath);
-      for (Asset asset : tx.browseAssets(tx.findBucket(getRepository()))) {
-        packagesBuilder.append(asset);
-      }
-      CompressorStreamFactory compressorStreamFactory = new CompressorStreamFactory();
-      ByteArrayOutputStream os = new ByteArrayOutputStream();
-      try (CompressorOutputStream cos = compressorStreamFactory.createCompressorOutputStream(GZIP, os)) {
-        try (OutputStreamWriter writer = new OutputStreamWriter(cos, UTF_8)) {
-          for (Entry<String, Map<String, String>> eachPackage : packagesBuilder.getPackageInformation().entrySet()) {
-            Map<String, String> packageInfo = eachPackage.getValue();
-            InternetHeaders headers = new InternetHeaders();
-            headers.addHeader(P_PACKAGE, packageInfo.get(P_PACKAGE));
-            headers.addHeader(P_VERSION, packageInfo.get(P_VERSION));
-            headers.addHeader(P_DEPENDS, packageInfo.get(P_DEPENDS));
-            headers.addHeader(P_IMPORTS, packageInfo.get(P_IMPORTS));
-            headers.addHeader(P_SUGGESTS, packageInfo.get(P_SUGGESTS));
-            headers.addHeader(P_LICENSE, packageInfo.get(P_LICENSE));
-            headers.addHeader(P_NEEDS_COMPILATION, packageInfo.get(P_NEEDS_COMPILATION));
-            Enumeration<String> headerLines = headers.getAllHeaderLines();
-            while (headerLines.hasMoreElements()) {
-              String line = headerLines.nextElement();
-              writer.write(line, 0, line.length());
-              writer.write('\n');
-            }
-            writer.write('\n');
-          }
-        }
-      }
-      return new Content(new BytesPayload(os.toByteArray(), "application/x-gzip"));
-    }
-    catch (CompressorException | IOException e) {
-      throw new RException(packagesPath, e);
-    }
-  }
-
   @Override
   @TransactionalTouchBlob
-  public Content getArchive(final String archivePath) {
-    checkNotNull(archivePath);
+  public Content getStoredContent(final String contentPath) {
```

Comment:
maybe combine buildPackageGz and putPackageGz into one method. Something like generateMetadata(basePath)

Reply:
done in e2e74ca6aee892191a160c1bd92155a5bd057a9c